### PR TITLE
Added Dockerfile

### PR DESCRIPTION
**Pull request by marcinguy**
Hi,

PR request to add Dockerfile (adding also GCC 10 and CLANG/LLVM 12 - not needed)

Helps to use it.

Thanks,

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
+FROM ubuntu:20.04 AS fuzz-introspector
```
Comment (DavidKorczynski) [inline Dockerfile:1]: Could you add the proper license?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 0
+RUN update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-10 0
+
+ENV LLVM_CONFIG=llvm-config-12
+
+RUN apt update
+RUN apt-get install python3.8-venv -y
+RUN apt-get install texinfo cmake -y
+WORKDIR /
+RUN git clone https://github.com/ossf/fuzz-introspector
```
Comment (DavidKorczynski) [inline Dockerfile:51]: Instead of cloning from github, could you do it so it uses the local source?

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
Thanks a lot! A few comments for now

File: Dockerfile (hunk 1 above)
Comment (marcinguy) [reply Dockerfile:1]: @DavidKorczynski Added. Or it should be a different one?

File: Dockerfile (hunk 2 above)
Comment (marcinguy) [reply Dockerfile]: Changed to work with local source.

File: Dockerfile (hunk 1 above)
Comment (DavidKorczynski) [reply Dockerfile:1]: looks good but please update it to 2022

Comment (marcinguy) [reply Dockerfile:1]: Done

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
Does this Dockerfile install a lot more packages than is actually needed? I would prefer if it doesn't install a lot of stuff that is not needed to run fuzz-introspector.

I have one locally as follows:
```
FROM ubuntu:20.04

ENV DEBIAN_FRONTEND noninteractive
RUN apt-get update && \
    apt-get upgrade -y && \
    apt-get install -y libc6-dev binutils && \
    apt-get autoremove -y

RUN apt-get update && \
    apt-get install -y software-properties-common && \
    add-apt-repository ppa:git-core/ppa && \
    apt-get update && \
    apt-get install -y \
        binutils-dev \
        build-essential \
        curl \
        git \
        jq \
        autoconf bison texinfo cmake

RUN apt-get install -y \
    python3 \
    python3-pip \
    python3.8-venv \
    clang-12

COPY . /fuzz-introspector
WORKDIR /fuzz-introspector

# First install python virtual environment
RUN python3 -m venv .venv && \
    . .venv/bin/activate && \
    pip install -r requirements.txt && \
    ./build_all.sh
```
Although it's not tested fully so am unsure about the completeness of it, however, it has much fewer packages installed

WDYT?

**Comment by marcinguy** [comment]
Both are fine to me.

Took a time to build yours. It builds.

However, it happens to be bigger than mine, I think also installs more packages:

fuzz-introspector  - my Dockerfile
fuzz-introspector2 - your paste  

```
REPOSITORY                               TAG          IMAGE ID       CREATED         SIZE
fuzz-introspector2                       Dockerfile   8fae482a728e   6 minutes ago   85GB
fuzz-introspector                        Dockerfile   05e8331a6515   2 days ago      83.6GB
```
Maybe you can add:

`--no-install-suggests --no-install-recommends`

Which should make it smaller & faster.

Don't have time to test it however.

Any Dockerfile would help people, since before it wasn't there.

Also I would not copy the source to image/container (`COPY . /fuzz-introspector`). I would mount the local dir and execute container with the mount.

Great work on fuzz-introspector.

Hope that helps.

**Comment by marcinguy** [comment]
Well you will have the folder in the Docker and it will work on it. When Docker exists/stops your results are outside of the container. AFAIK You might don't even need to start container to use binaries in the folder (depending on linkage etc)

In your setup you need to start container and enter it. Somehow feed your project there (via mount or SSH copy etc).

Whatever works :) Not an expert on Docker.

**Comment by DavidKorczynski** [comment]
sounds good, thanks! Am not sure I understand the reasoning about packages though, is your thought that mine installs more packages because of the size? My reasoning came from your Dockerfile having a lot of packages installed by way of `apt`. I think perhaps the size difference comes from me copying in more, e.g. my `oss-fuzz` folder is pretty sizeable within the `oss_fuzz_integration`.

In either case let's get this merged and we'll take it from there! Thanks again!

**Review by DavidKorczynski — APPROVED** [review]

**Comment by marcinguy** [comment]
Thanks. I can try your Dockerfile again, but add `--no-install-suggests --no-install-recommends` to `apt` and see if it still works. Since I think your does install suggested and recommended packages by default. As I understand they're typically not required. I think I saw many other packages being added that way. 

This could be the most optimal sizewise. 

I get back as I know.

**Pull request merged**